### PR TITLE
#345 Spacing variables for the css usage

### PR DESCRIPTION
**Pull request by marcinsawicki**
Resolves: #345 

## Description
Scss variables of spacing, to the use in styling for components.

## Storybook
- [Storybook](https://613a8e945a5665003a05113b-czjyoqmqne.chromatic.com/?path=/story/foundations-spacing--page)

## Checklist

**Obligatory:**

- [x] Self-review
- [ ] Unit & integration tests
- [x] Storybook cases
- [ ] Design review
- [ ] Functional (QA) review

**Optional:**

- [ ] Accessibility cases (keyboard control, correct HTML markup, etc.)

### CODE REVIEW

**Comment by sgraczyk** [comment]
Why aren't we using `CSS variables` for that, or at least both with SCSS?

File: packages/react-components/.storybook/preview.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import '../src/themes/legacy.scss';
 import '../src/themes/light.scss';
 import '../src/themes/dark.scss';
+import '../src/spacing/spacing.scss';
```
Comment (sgraczyk) [inline packages/react-components/.storybook/preview.js:4]: Do we need `scss` at all? Using `CSS variables` feels convenient

Comment (sgraczyk) [reply packages/react-components/.storybook/preview.js:4]: Ok, nevermind it's just an extension of file, it still uses CSS vars

File: packages/react-components/src/index.scss [hunk 2]
```diff
@@ -1,4 +1,7 @@
 @use './themes/legacy';
+@use './themes/light';
+@use './themes/dark';
+@use './spacing/spacing';
```
Comment (sgraczyk) [inline packages/react-components/src/index.scss:4]: how about?:
```suggestion
@use './foundations/spacing';
```

File: packages/react-components/src/index.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import './index.scss';
 
 export { DesignTokens } from './themes/designTokens';
+export { SpacingTokens } from './spacing/spacingTokens';
```
Comment (sgraczyk) [inline packages/react-components/src/index.ts:4]: Enumerations (even realized as consts) should have a singular form.
```suggestion
export { SpacingToken } from './foundation/spacing-tokens';
```

File: packages/react-components/src/index.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import './index.scss';
 
 export { DesignTokens } from './themes/designTokens';
```
Comment (sgraczyk) [inline packages/react-components/src/index.ts:3]: We could fix this also
```suggestion
export { DesignToken } from './themes/design-oken';
```

File: packages/react-components/src/spacing/spacing.scss [hunk 5]
```diff
@@ -0,0 +1,16 @@
+:root {
```
Comment (sgraczyk) [inline packages/react-components/src/spacing/spacing.scss:1]: @kstvsko @martagranisz Are we able to confirm that `spacing` tokens should be `theme-agnostic`? What I mean is if we are sure that there will be no `theme` that could introduce different spacings?

File: packages/react-components/src/spacing/spacingTokens.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+export const SpacingTokens = {
```
Comment (sgraczyk) [inline packages/react-components/src/spacing/spacingTokens.ts:1]: Alternatively `SPACING_TOKENS`, hard to say what's our approach here, as it seems more of a `enumeration` than a set of data.
```suggestion
export const SpacingToken = {
```

File: packages/react-components/.storybook/preview.js (hunk 1 above)
Comment (marcinsawicki) [reply packages/react-components/.storybook/preview.js:4]: 👍

File: packages/react-components/src/index.ts (hunk 4 above)
Comment (marcinsawicki) [reply packages/react-components/src/index.ts:3]: 👍 But we need to remember to include this as (small) breaking change in release docs, as we use `DesignTokens` enum in another app.

File: packages/react-components/src/spacing/spacing.scss (hunk 5 above)
Comment (sgraczyk) [reply packages/react-components/src/spacing/spacing.scss]: For now we assumed spacings are constant for all themes

**Review by sgraczyk — APPROVED** [review]

**Review by kstvsko — APPROVED** [review]
I think everything is right, my only concern is the **Examples** in the Foundations/Spacing. I think we should address this during our new approach to the documentation to be more descriptive and clear, I will note it down.

**Pull request merged**